### DISCUSSION TITLE
Using custom payload pool from DDS layer

### DISCUSSION THREAD

**Discussion by OhadMeir** [eProsima/Fast-DDS#2931] [Ideas]
Hi, I want to receive large messages from a UDP transport

As I understand there are 2 copies of the data:
1. From the network stack to the DataReaderHistory
2. From the history to the user

The second copy can be avoided by using loans ([example](https://fast-dds.docs.eprosima.com/en/latest/fastdds/use_cases/zero_copy/zero_copy.html#datareader)), but it is not enough in my use case - a library that needs to copy the data to user provided buffers.

I want to copy directly to the user buffers by using a custom payload pool ([see here](https://fast-dds.docs.eprosima.com/en/latest/fastdds/rtps_layer/rtps_layer.html#using-a-custom-payload-pool)) but I can't find the way to pass that pool to the DDS DomainParticipant/Subscriber/DataReader.

Is it possible to use custom IPayloadPool implementations while using DDS and not creating the RTPSReader directly?

Thanks
Ohad

**Comment by MiguelCompany** [comment]
@OhadMeir This is currently not possible, but would indeed be a nice addition to the API.

I have moved the discussion into the Ideas category, where we are keeping all feature requests.

**Reply by OhadMeir** [reply]
Thank you
